jerry-core/ecma/base/ecma-helpers.c

```diff
@@ -166,7 +166,7 @@ ecma_create_object_lex_env (ecma_object_t *outer_lexical_environment_p, /**< out
  * @return true  - if object is a lexical environment
  *         false - otherwise
  */
-inline bool JERRY_ATTR_PURE
+extern JERRY_ALWAYS_INLINE bool JERRY_ATTR_PURE
```

Comment:
This is not an identical change. This function is simply `inline`, not "always inline". (Whether that's right, that can be discussed ofc.)

(This holds for several functions in this module.)

Reply:
inline have same effect with __forceinline in MSVC, so the JERRY_ALWAYS_INLINE macro are needed to make sure use always prefix JERRY_ALWAYS_INLINE with static or extern.
And if the function have no need inline, we can always not use?
What's the choose between inline and always_inline?

Reply:
I'm happy to call in other maintainers of the project to discuss why some functions are simply inline while others are always inline. However, if we cannot dig to the roots of that design decision now - which can easily happen - then we should not introduce non-identical changes with potential unknown side-effects just because of MSVC.

If I get it correctly, we could
- simply keep the `inline` keyword everywhere,
- add `extern` keyword to non-static functions where that was missing (I'm OK with that addition to make MSVC happy),
- keep `JERRY_ATTR_ALWAYS_INLINE` macro name, but change its placement in the function signature, and
- define `JERRY_ATTR_ALWAYS_INLINE` for MSVC as `__forceinline`.

And this would be the least disruptive approach then. Would that work?

Reply:
make sense, done with your suggestion